### PR TITLE
Calculating electric potential due to a point charge using lambda functions and maps.

### PR DESCRIPTION
**Pull request by Tolani4**
This Pull Request implements a Python function to calculate the electric potential V at various distances due to a point charge. The calculation is based on Coulomb's Law and uses functional programming concepts, lambda functions, list comprehension, and map() function for implementation.

### CODE REVIEW

**Comment by Tolani4** [comment]
After submitting my pull requests I'm still receiving errors from github specifically pylint. Am I just supposed to apply the changes it recommends such as " C0301: Line too long (108/100) (line-too-long)"

**Comment by laserlab** [comment]
> After submitting my pull requests I'm still receiving errors from github specifically pylint. Am I just supposed to apply the changes it recommends such as " C0301: Line too long (108/100) (line-too-long)"

yes, since the reviewer will point it out and we want to write compliant code

**Comment by laserlab** [comment]
Though your low score (1.3/10) most likely stems from more serious issues like: 
> Electric_potential_due_to_point_charge.py/Electric_potential.py:30:21: C3001: Lambda expression assigned to a variable. Define a function using the "def" keyword instead. (unnecessary-lambda-assignment)

which I discussed in class as we talked about lambda functions

**Comment by Tolani4** [comment]
Okay thank you. For the "missing module docstring" error. Do I just need to create a separate file called docstring and just explain my code and the physics behind it? and for the error "C0115: Missing class docstring (missing-class-docstring)",  Do I just create a class called docstring in that file? Just trying to understand what the errors mean.

**Comment by laserlab** [comment]
As we discussed in class docstring explanation can be found in the lecture slides in hands on examples and here:

https://peps.python.org/pep-0257/

**Comment by laserlab** [comment]
@ojha-aditya Please review asap

**Comment by ojha-aditya** [comment]
Can I as reviewer resolve this notification about the branch being out-of-date or we simply mention it in the review as a todo?

**Comment by laserlab** [comment]
> Also, do we need to have a main function in the module?

no, since it’s a module

**Comment by laserlab** [comment]
> Can I as reviewer resolve this notification about the branch being out-of-date or we simply mention it in the review as a todo?

you an mention it

File: Electric_potential_due_to_point_charge.py/Electric_potential.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+# Constants
+k = 8.99e9  # Coulomb's constant in Nm^2/C^2
+
+def electric_potential(charge, distances):
+    """
+    Calculate the electric potential due to a point charge at multiple distances.
+
+    Parameters:
+    charge (float): The charge in Coulombs.
```
Comment (laserlab) [inline Electric_potential_due_to_point_charge.py/Electric_potential.py:23]: if you do not require nor check for float do not use it in docstring

**Review by laserlab — CHANGES_REQUESTED** [review]
- no .py in folder name
- data type in docstring
- tons of linting errors

**Comment by ojha-aditya** [comment]
We should not mention the things you have already pointed in our review @laserlab should we?

**Comment by laserlab** [comment]
> We should not mention the things you have already pointed in our review @laserlab should we?

Please mention all you noticed, independent from what I comment.

**Review by ojha-aditya — CHANGES_REQUESTED** [review]
Along with the ones already pointed out, here is a good explanation for the unnecessary-lambda-assignment pylint message:

https://pylint.pycqa.org/en/latest/user_guide/messages/convention/unnecessary-lambda-assignment.html
https://www.geeksforgeeks.org/overuse-of-lambda-expressions-in-python/

**Review by ojha-aditya — CHANGES_REQUESTED** [review]
The binary file needs to be removed as well.

**Comment by laserlab** [comment]
no fixes, not merging

**Pull request closed without merging**